src/Service/ApiRequest.php

```diff
@@ -0,0 +1,29 @@
+<?php
+/**
+ * Created by PhpStorm.
+ * User: chris
+ * Date: 18/12/18
+ * Time: 11:11
+ */
+
+namespace App\Service;
+
+use GuzzleHttp\Client;
+use App\Entity\Simulator;
+
+class ApiRequest
+{
+    /**
+     * @param $simulator
+     * @throws \GuzzleHttp\Exception\GuzzleException
+     */
+    public function dataLoad(Simulator $simulator)
+    {
+        $client = new Client();
+        $res = $client->request('GET', 'https://api-adresse.data.gouv.fr/search/?q=\''.$simulator->getCity().'\'');
+        $json = $res->getBody();
+        $data = json_decode($json, true);
+
+        return $data;
+    }
```

Comment:
j'aurai ajouté une méthode getCity qui récupèrerai la city plutot que de le faire dans le controlleur.
Tu peux avoir un construct avec dedans
$this->data = $this->dataLoad();

et dans ta methode getCity()
{
    return $this->data = $this->dataLoad()['features'][0]['properties']['city'] 
}